### ISSUE TITLE
Programs assume files from nix store writable

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#55683]
## Issue description

Some programs have issues on NixOS because files copied from nix store are read only.

This results in permission errors like this:

`PermissionError: [Errno 13] Permission denied: '/home/davidak/.openshot_qt/title/temp.svg' `

Is there any way to solve it for us or do we rely on upstream to fix it? What can we suggest them as a better way doing it?

I had this multiple times: https://github.com/NixOS/nixpkgs/issues/32898 https://github.com/NixOS/nixpkgs/issues/43685

Some developers refuse to fix such errors: "Both 1970 dates and invalid permissions are caused by NixOS being crazy, not Nikola."

**Comment by FlorianFranzen** [comment]
Have you tried just changing the files owner and permission with:
```
sudo chown davidak ~/.openshot_qt/title/temp.svg
chmod +w ~/.openshot_qt/title/temp.svg
```

**Comment by davidak** [comment]
@FlorianFranzen No. That would not solve the problem in general.

**Comment by erictapen** [comment]
> Some developers refuse to fix such errors: "Both 1970 dates and invalid permissions are caused by NixOS being crazy, not Nikola."

Only solution I see here is that e.g. you provide a patch for the package that takes care of the file permissions. If upstream doesn't want it, it would need to be maintained in Nixpkgs.

**Comment by lifranc** [comment]
> "Both 1970 dates and invalid permissions are caused by NixOS being crazy, not Nikola."

That's wrong; it stems from unixfs being stupid.

**Comment by peti** [comment]
I don't believe that we can do anything about this other than reporting the issue upstream, hoping that they'll fix it (as the OpenShot developers have in d7bbcdf274eaa3de1f813f391b64626b2f87bc49). If they don't want to do that, then we'll have to do it ourselves.

Anyhow, would anyone object to closing this bug report? It does not seem to represent an actionable task and therefore it's to tell what to do about it.

**Comment by davidak** [comment]
>I don't believe that we can do anything about this other than reporting the issue upstream

Could we add it to the documentation with an example? Maybe in a Troubleshooting section of the nixpkgs manual.

Then others don't have to open the same issue again.

**Comment by peti** [comment]
> Could we add it to the documentation with an example? 

Would you like to create a PR with a suggestion?

> Then others don't have to open the same issue again.

My guess is that others *have* to open a new ticket every time they run into an issue like this. There is no other way to fix them.

**Comment by davidak** [comment]
>Would you like to create a PR with a suggestion?

No, i just can't use the documentation system. I'm waiting for the migration to Asciidoc!

>My guess is that others have to open a new ticket every time they run into an issue like this. There is no other way to fix them.

OK, but it would help to have documented how to fix them.

Just explain the issue in one sentence and "report the issue upstream". And when upstream don't wan to fix it, maintain a patch in nixpkgs.

**Comment by peti** [comment]
> Just explain the issue in one sentence and "report the issue upstream".

No, I don't care enough to extend the docs for this issue; please do it yourself if you believe that this is important.

**Comment by davidak** [comment]
@peti sorry, i don't mean you should do it! i just argue that we should leave the issue open until the information is documented by someone. that someone might be me in half a year when writing documentation is easier and don't involve writing XML.

**Comment by davidak** [comment]
...

**Comment by davidak** [comment]
...

**Comment by khaled** [comment]
...
